From 2b27f0eb5858a4fd296170fb7e4533e5fc7aa3e9 Mon Sep 17 00:00:00 2001
From: Xisco Fauli <xiscofauli@libreoffice.org>
Date: Fri, 21 Jun 2024 11:58:36 +0200
Subject: [PATCH] libxml2: upgrade to 2.13.1

Building with --without-system-libxml fails with
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPScanProxy'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPClose'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPReturnCode'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPRead'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPCleanup'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPMethod'
/usr/bin/ld: /home/xisco/libreoffice/workdir/UnpackedTarball/raptor/src/.libs/libraptor2.so: undefined reference to `xmlNanoHTTPInit'
collect2: error: ld returned 1 exit status

so use --without-www in external/redland/ExternalProject_raptor.mk
since it seems no to be used by raptor.
Libxml2 disabled http support by default in
https://gitlab.gnome.org/GNOME/libxml2/-/commit/3018842c07e9b88d6bb0257f5644e7695cdeb2e2

Downloaded from https://download.gnome.org/sources/libxml2/2.13/libxml2-2.13.1.tar.xz

Kudos to Michael Weghorn for helping me with this patch

Change-Id: I2e41021e8aea3551eb9eec29cf12f6cd9f6ff89e
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/169327
Reviewed-by: Taichi Haradaguchi <20001722@ymail.ne.jp>
Tested-by: Jenkins
---
 download.lst                                      | 6 +++---
 external/libxml2/ExternalPackage_libxml2.mk       | 2 +-
 external/libxml2/libxml2-XMLCALL-redefine.patch.0 | 8 ++++----
 external/libxml2/libxml2-icu-sym.patch.0          | 8 ++++----
 external/libxml2/libxml2-icu.patch.0              | 8 ++++----
 external/redland/ExternalProject_raptor.mk        | 2 +-
 6 files changed, 17 insertions(+), 17 deletions(-)

diff --git a/download.lst b/download.lst
index 5ba08be058977..1b4ebfb4bbe93 100644
--- a/download.lst
+++ b/download.lst
@@ -424,9 +424,9 @@ XMLSEC_TARBALL := xmlsec1-1.3.4.tar.gz
 # three static lines
 # so that git cherry-pick
 # will not run into conflicts
-LIBXML_SHA256SUM := 43ad877b018bc63deb2468d71f95219c2fac196876ef36d1bee51d226173ec93
-LIBXML_VERSION_MICRO := 8
-LIBXML_TARBALL := libxml2-2.12.$(LIBXML_VERSION_MICRO).tar.xz
+LIBXML_SHA256SUM := 25239263dc37f5f55a5393eff27b35f0b7d9ea4b2a7653310598ea8299e3b741
+LIBXML_VERSION_MICRO := 1
+LIBXML_TARBALL := libxml2-2.13.$(LIBXML_VERSION_MICRO).tar.xz
 # three static lines
 # so that git cherry-pick
 # will not run into conflicts
diff --git a/external/libxml2/ExternalPackage_libxml2.mk b/external/libxml2/ExternalPackage_libxml2.mk
index 799044575f9a3..17190c77fd185 100644
--- a/external/libxml2/ExternalPackage_libxml2.mk
+++ b/external/libxml2/ExternalPackage_libxml2.mk
@@ -21,7 +21,7 @@ else # COM=MSC
 $(eval $(call gb_ExternalPackage_add_file,libxml2,$(LIBO_URE_LIB_FOLDER)/libxml2.dll,win32/bin.msvc/libxml2.dll))
 endif
 else # OS!=WNT
-$(eval $(call gb_ExternalPackage_add_file,libxml2,$(LIBO_URE_LIB_FOLDER)/libxml2.so.2,.libs/libxml2.so.2.12.$(LIBXML_VERSION_MICRO)))
+$(eval $(call gb_ExternalPackage_add_file,libxml2,$(LIBO_URE_LIB_FOLDER)/libxml2.so.2,.libs/libxml2.so.2.13.$(LIBXML_VERSION_MICRO)))
 endif
 endif # DISABLE_DYNLOADING
 
diff --git a/external/libxml2/libxml2-XMLCALL-redefine.patch.0 b/external/libxml2/libxml2-XMLCALL-redefine.patch.0
index d9ca23be7ce49..57d23d885943d 100644
--- a/external/libxml2/libxml2-XMLCALL-redefine.patch.0
+++ b/external/libxml2/libxml2-XMLCALL-redefine.patch.0
@@ -1,11 +1,11 @@
 --- include/libxml/xmlexports.h	2023-05-24 12:48:46.179570708 +0100
 +++ include/libxml/xmlexports.h	2023-05-24 12:48:56.563577488 +0100
-@@ -38,8 +38,6 @@
+@@ -31,8 +31,6 @@
+ #define XMLPUBVAR XMLPUBLIC extern
  
- /** DOC_DISABLE */
  /* Compatibility */
 -#define XMLCALL
 -#define XMLCDECL
- #if !defined(LIBXML_DLL_IMPORT)
- #define LIBXML_DLL_IMPORT XMLPUBVAR
+ #ifndef LIBXML_DLL_IMPORT
+   #define LIBXML_DLL_IMPORT XMLPUBVAR
  #endif
diff --git a/external/libxml2/libxml2-icu-sym.patch.0 b/external/libxml2/libxml2-icu-sym.patch.0
index 07676228da194..d89a1c4c544be 100644
--- a/external/libxml2/libxml2-icu-sym.patch.0
+++ b/external/libxml2/libxml2-icu-sym.patch.0
@@ -3,14 +3,14 @@ Find bundled ICU in workdir and use debug .libs when needed
 diff -up win32/Makefile.msvc.dt win32/Makefile.msvc
 --- win32/Makefile.msvc.dt	2014-07-18 19:00:23.372103963 +0200
 +++ win32/Makefile.msvc	2014-07-18 19:01:39.347982929 +0200
-@@ -46,6 +46,7 @@ CPPFLAGS = $(CPPFLAGS) /D "_REENTRANT"
+@@ -42,6 +42,7 @@
  CC = cl.exe
  CFLAGS = /nologo /D "_WINDOWS" /D "_MBCS" /D "NOLIBTOOL" /W3 /wd4244 /wd4267 $(CRUNTIME)
  CFLAGS = $(CFLAGS) /I$(XML_SRCDIR) /I$(XML_SRCDIR)\include /I$(INCPREFIX)
 +CFLAGS = $(CFLAGS) /I$(WORKDIR)/UnpackedTarball/icu/source/i18n /I$(WORKDIR)/UnpackedTarball/icu/source/common
- !if "$(WITH_THREADS)" != "no"
- CFLAGS = $(CFLAGS) /D "_REENTRANT"
- !endif
+ !if "$(WITH_THREADS)" == "ctls"
+ CFLAGS = $(CFLAGS) /D "XML_THREAD_LOCAL=__declspec(thread)"
+ !else if "$(WITH_THREADS)" == "posix"
 @@ -62,7 +63,9 @@
  # The linker and its options.
  LD = link.exe
diff --git a/external/libxml2/libxml2-icu.patch.0 b/external/libxml2/libxml2-icu.patch.0
index 84197763f878e..85ce8b3aba2ba 100644
--- a/external/libxml2/libxml2-icu.patch.0
+++ b/external/libxml2/libxml2-icu.patch.0
@@ -3,14 +3,14 @@ Find bundled ICU in workdir and use debug .libs when needed
 diff -up win32/Makefile.msvc.dt win32/Makefile.msvc
 --- win32/Makefile.msvc.dt	2014-07-18 19:00:23.372103963 +0200
 +++ win32/Makefile.msvc	2014-07-18 19:01:39.347982929 +0200
-@@ -45,6 +45,7 @@ CPPFLAGS = $(CPPFLAGS) /D "_REENTRANT"
+@@ -42,6 +42,7 @@
  CC = cl.exe
  CFLAGS = /nologo /D "_WINDOWS" /D "_MBCS" /D "NOLIBTOOL" /W3 /wd4244 /wd4267 $(CRUNTIME)
  CFLAGS = $(CFLAGS) /I$(XML_SRCDIR) /I$(XML_SRCDIR)\include /I$(INCPREFIX)
 +CFLAGS = $(CFLAGS) /I$(WORKDIR)/UnpackedTarball/icu/source/i18n /I$(WORKDIR)/UnpackedTarball/icu/source/common
- !if "$(WITH_THREADS)" != "no"
- CFLAGS = $(CFLAGS) /D "_REENTRANT"
- !endif
+ !if "$(WITH_THREADS)" == "ctls"
+ CFLAGS = $(CFLAGS) /D "XML_THREAD_LOCAL=__declspec(thread)"
+ !else if "$(WITH_THREADS)" == "posix"
 @@ -67,6 +68,7 @@ CFLAGS = $(CFLAGS) $(SOLARINC)
  # The linker and its options.
  LD = link.exe
diff --git a/external/redland/ExternalProject_raptor.mk b/external/redland/ExternalProject_raptor.mk
index 8ac05880766b3..5bfab78deba80 100644
--- a/external/redland/ExternalProject_raptor.mk
+++ b/external/redland/ExternalProject_raptor.mk
@@ -30,7 +30,7 @@ $(call gb_ExternalProject_get_state_target,raptor,build):
 		CPPFLAGS="$(if $(SYSBASE),-I$(SYSBASE)/usr/include) $(gb_EMSCRIPTEN_CPPFLAGS)" \
 		$(gb_RUN_CONFIGURE) ./configure --disable-gtk-doc \
 			 --enable-parsers="rdfxml ntriples turtle trig guess rss-tag-soup" \
-			--with-www=xml \
+			--without-www \
 			--without-xslt-config \
 			$(gb_CONFIGURE_PLATFORMS) \
 			$(if $(CROSS_COMPILING),$(if $(filter INTEL ARM,$(CPUNAME)),ac_cv_c_bigendian=no)) \
